src/main/java/de/kaleidox/e2uClaim/config/PlayerPreferenceManager.java

```diff
@@ -0,0 +1,69 @@
+package de.kaleidox.e2uClaim.config;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import de.kaleidox.e2uClaim.E2UClaim;
+import de.kaleidox.e2uClaim.interfaces.Initializable;
+import de.kaleidox.e2uClaim.interfaces.Terminatable;
+
+import org.bukkit.configuration.file.FileConfiguration;
+
+public enum PlayerPreferenceManager implements Initializable, Terminatable {
+    INSTANCE;
+
+    private final Map<UUID, Map<String, String>> playerPropertyValueMap = new ConcurrentHashMap<>();
+    private final Map<String, String> defaultValues = new ConcurrentHashMap<>();
+
+    @Override
+    public void init() throws Throwable {
+        FileConfiguration config = E2UClaim.getConfig("userPreferences");
+        FileConfiguration defaults = E2UClaim.getConfig("config");
+        defaultValues.put("autoLock", Boolean.toString(defaults.getBoolean("defaults.auto-lock")));
+        defaultValues.put("toolItem", defaults.getString("defaults.tool-item"));
+        Objects.requireNonNull(config.getKeys(false)).forEach(key -> {
+            UUID uniqueId = UUID.fromString(key);
+            Map<String, String> map = new ConcurrentHashMap<>();
+            Objects.requireNonNull(config.getConfigurationSection(key)).getKeys(false).forEach(property -> {
+                String value = config.getString(property);
+                map.put(property, value);
+            });
+            playerPropertyValueMap.put(uniqueId, map);
+        });
+    }
+
+    @Override
+    public void terminate() throws IOException {
+        FileConfiguration config = E2UClaim.getConfig("userPreferences");
+        for (Map.Entry<UUID, Map<String, String>> entry : playerPropertyValueMap.entrySet()) {
+            String uuid = entry.getKey().toString();
+            Map<String, String> values = entry.getValue();
+            values.forEach((key, value) -> E2UClaim.getConfig("userPreferences").set(uuid + "." + key, value));
```

Comment:
It would be nicer to store the `userPreferences` configuration object in a local variable instead of calling the method every time.

Reply:
This like we won't set it inside the configuration, but in the variable only. Thats why i do it like that here.

Reply:
Having it this way will require a `Map#compute` call for every available variable. A local variable would be quicker.

Reply:
Dude, do you know how seting something works? when you make something to a variable, it makes a copy of it a.ka you dont set it inside the one you should, you set it inside the copy